### PR TITLE
1952 cleanup app settings

### PR DESCRIPTION
**Pull request by timabell**
### Context

We were learning dotnet core as we went, it has too many ways to configure everything.

### Changes proposed in this pull request

Get as close as possible to running the tests out of the box.

To make this app easier to run and the config system easier to follow.

* Refactor - inline all the config keys
* Set default database configuration
* Moved all the defaults into `appsetting.json`.
* Made test suite read `appsettings.json`.
* Used same dev credentials as manage-courses-backend because everyone
  has mcbe which sets up this user.
* Moved config into Config folder to be more like rails.
* Expanded default config to sensible defaults.
* Stripped dev config back to only changing debug level.
* Kill mcconfig validator - more pain than gain
* Use our config builder for main app instead of implicit one from app builder.

### Guidance to review

```
git clone
```

```
cd tests\ManageCourses.Tests\
dotnet user-secrets list
dotnet test
```

```
cd src/ManageCourses.Api
dotnet user-secrets list
dotnet run
```

ask me for sign-in config

See if it behaves in qa when merged.

### CODE REVIEW

**Comment by timabell** [comment]
[Update c# to use new SEND attribute](https://trello.com/c/DnoNVabp/1953-update-c-to-use-new-send-attribute)

File: src/ManageCourses.Api/ManageCourses.Api.csproj [hunk 1]
```diff
@@ -28,13 +28,18 @@
     <ProjectReference Include="..\ManageCourses.Domain\ManageCourses.Domain.csproj" />
   </ItemGroup>
   <ItemGroup>
-    <Content Update="appsettings.json">
+    <None Update="app_offline.htm.example">
```
Comment (defong) [inline src/ManageCourses.Api/ManageCourses.Api.csproj]: Why?

Comment (timabell) [reply src/ManageCourses.Api/ManageCourses.Api.csproj]: moved from below because csproj is unstable, meh

**Comment by timabell** [comment]
wrong branch name. new PR #327

**Pull request closed without merging**